---
layout: post
title:  "파이썬 24. 파이썬 에러메시지(try, escept, else, finally"
description: "파이썬 에러메시지 다루기 try, except, else,  finally"
author: Byeonguk Kim
date:   2019-04-29 01:06:23 +0700
categories: [Python]
---

### 2019.08.07 파이썬 에러메시지(try, escept, else, finally

\# 질문에 답하기

1.

## try, except를 통한 에외처리

* 파이썬에서 특정 구문을 실행하고 싶은데, 혹시라도 오류가 날까 두려울 때.
* 어떤 오류가 날지 모를 때
* try, except를 통해서 해당 문제를 해결할 수 있다.


## try, except

```
try:
    print('이것부터 실행해봅니다.')

except:
    print('위에 try 구문이 안될 시 이것을 실행합니다')

```

* 원래는 except 뒤에 일어날 에러명을 적어주면 된다.
* 하지만 단순히 예외처리를 위한 것이라면 굳이 적어주지 않아도 되는 것 같다.

## 특정 경우 에러 일으키키 (raise)

* raise를 통해 에러를 일으킬 수 있다.
* try, except을 통해서 좀 더  효율적으로 쓸 수 있다.

```
try:
    if err_code != 0:
        raise Exception

except:
    print('err_code가 0이 아닙니다')
```

* 해당 부분이 아니면 raise Exception 에러를 일으켜서 except로 넘어 간다.

## 파이썬 에러와 예외에 대한 좋은 자료

* [8. 에러와 예외 — Python 3.7.4 문서](https://docs.python.org/ko/3/tutorial/errors.html)


## try와 except, else, finally를 통한 예시

```
def divide(x, y):
    try:
        result = x / y
    except:
        print('0으로 나눌수가 없어요')
        return False
    else:
        print('결과: ', result)
        return True
    finally:
        print('나누기  연산 종료')
```    
    
* try를 해보고 가능하면 else로 넘어가서  해당 결과값에 대해 출력한다.
* finally는 항상 끝나기 전에 무조건 실행된다.
* finally를 잘 활용하면 왠지 더 좋은 결과물을 낼 수 있을 것  같다.

![12A9EF95-FE55-4858-AAAE-AD6579CB8370](https://user-images.githubusercontent.com/46436843/62585797-8044c680-b8f5-11e9-92d8-6217f4f40033.png)


* 출처 : [[Python] 예외처리 - try, except, else, finally](https://gomguard.tistory.com/122)
